clients/src/main/java/org/apache/kafka/common/requests/CreateTopicRequest.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    public CreateTopicRequest(Struct struct) {
+        super(struct);
+
+        Object[] requestStructs = struct.getArray(REQUESTS_KEY_NAME);
+        topics = new HashMap<String, TopicDetails>();
+        for (Object requestStructObj : requestStructs) {
+            Struct singleRequestStruct = (Struct) requestStructObj;
+            String topic = singleRequestStruct.getString(TOPIC_KEY_NAME);
+            int partitions = singleRequestStruct.getInt(PARTITIONS_KEY_NAME);
+            int replicationFactor = singleRequestStruct.getInt(REPLICATION_FACTOR_KEY_NAME);
+
+            //replica assignment
+            Object[] assignmentsArray = singleRequestStruct.getArray(REPLICA_ASSIGNMENT_KEY_NAME);
+            Map<Integer, List<Integer>> partitionReplicaAssignments = new HashMap<Integer, List<Integer>>(assignmentsArray.length);
+            for (Object assignmentStructObj : assignmentsArray) {
+                Struct assignmentStruct = (Struct) assignmentStructObj;
+
+                Integer partitionId = assignmentStruct.getInt(REPLICA_ASSIGNMENT_PARTITION_ID_KEY_NAME);
+
+                Object[] replicasArray = assignmentStruct.getArray(REPLICA_ASSIGNMENT_REPLICAS_KEY_NAME);
+                List<Integer> replicas = new ArrayList<Integer>(replicasArray.length);
+                for (Object replica : replicasArray) {
+                    replicas.add((Integer) replica);
+                }
+
+                partitionReplicaAssignments.put(partitionId, replicas);
+            }
+
+            Object[] configArray = singleRequestStruct.getArray(CONFIGS_KEY_NAME);
+            Map<String, String> configs = new HashMap<String, String>(configArray.length);
+            for (Object configStructObj : configArray) {
+                Struct configStruct = (Struct) configStructObj;
+
+                String key = configStruct.getString(CONFIG_KEY_KEY_NAME);
+                String value = configStruct.getString(CONFIG_VALUE_KEY_NAME);
+
+                configs.put(key, value);
+            }
+
+            TopicDetails args = new TopicDetails(
+                partitions, replicationFactor, partitionReplicaAssignments, configs);
+            topics.put(topic, args);
+        }
+    }
+
+    @Override
+    public AbstractRequestResponse getErrorResponse(int versionId, Throwable e) {
+        Map<String, Errors> topicErrors = new HashMap<String, Errors>();
+        for (String topic : topics.keySet()) {
+            topicErrors.put(topic, Errors.forException(e));
+        }
+
+        switch (versionId) {
+            case 0:
+                return new CreateTopicResponse(topicErrors);
+            default:
+                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
+                    versionId, this.getClass().getSimpleName(), ProtoUtils.latestVersion(ApiKeys.CREATE_TOPIC.id)));
+        }
+    }
+
+    public Map<String, TopicDetails> topics() {
```

Comment:
Yes it's used to get the topics from the request: https://github.com/apache/kafka/pull/626/files#diff-d45970e44e2636ec847b63ac71827b71R913